tests/test_mask.py

```diff
@@ -0,0 +1,63 @@
+import sys
+import pytest
+import geopandas as gpd
+import numpy as np
+
+sys.path.append('../modules/')
+from mask import check_polygon, check_pre_proj, check_post_proj, check_coords
+
+world = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres'))
+can_index = world[world.name == "Canada"].index
+can_geom = world.loc[can_index, 'geometry']
+
+@pytest.mark.parametrize('p,passed', [
+    ({'string'}, False),
+    ({5}, False),
+    (can_geom, True),
+    (world, False),
+    (gpd.GeoSeries(), False)])
+def test_check_polygon(p, passed):
+    if passed:
+        assert check_polygon(p)
+    else:
+        with pytest.raises((TypeError, ValueError)):
+            check_polygon(p)
+
+@pytest.mark.parametrize('p,passed', [
+    ({'string'}, False),
+    ({5}, False),
+    (can_geom, True),
+    (world, False),
+    (gpd.GeoSeries(), False)])
+def test_check_pre_proj(p, passed):
+    if passed:
+        assert check_pre_proj(p)
+    else:
+        with pytest.raises((TypeError, ValueError)):
+            check_pre_proj(p)
+
+@pytest.mark.parametrize('p,passed', [
```

Comment:
If *all* of your test cases are supposed to fail, under what circumstances is it supposed to pass?

Reply:
In order to test if it passed, I need to provide a rotated shapefile to the test.

I was trying to avoid calling `rotate_shapefile` in `test_mask.py`.

To include this, should I import a correctly rotated shapefile in the `data` directory? What are the best practices for this? If I use `rotate_shapefile` then it will end up testing itself.

Reply:
And should
```python
world = gpd.read_file(gpd.datasets.get_path('naturalearth_lowres'))
can_index = world[world.name == "Canada"].index
can_geom = world.loc[can_index, 'geometry']
```

Also be replaced for a shapefile in the data directory?

Reply:
I've gone ahead and added shapefiles directly to the data directory in `climpyrical/tests/data`.